backtracking/generate_parentheses.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file
+ * @brief Generates all combinations of well-formed parentheses.
+ *
+ * @details a sequence of parentheses is well-formed if each opening parentheses
+ has a corresponding closing parenthesis
+ *  and the closing parentheses are correctly ordered
+ *
+ * @author [Giuseppe Coco](https://github.com/WoWS17)
+
+ */
+
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+class generate_parentheses {
+ private:
+    std::vector<std::string> res;  // Contains all possible valid patterns
```

Comment:
This is needed so that Doxygen can detect it.
```suggestion
    std::vector<std::string> res;  ///< Contains all possible valid patterns
```